smarts/sstudio/generators.py

```diff
@@ -332,3 +333,63 @@ def _resolve_log_dir(self, log_dir):
             log_dir = make_dir_in_smarts_log_dir("_duarouter_routing")
 
         return os.path.abspath(log_dir)
+    
+
+def tripwritexml(resolve_route, traffic, doc):
+        """Wrtes a traffic spec into a route file. Typically this would be the source
+        data to Sumo's DUAROUTER.
+        """
+            # Actors and routes may be declared once then reused. To prevent creating
+            # duplicates we unique them here.
+        for trip in traffic.trips:
+            actor = trip.actor
+            sigma = min(1, max(0, actor.imperfection.sample()))  # range [0,1]
+            min_gap = max(0, actor.min_gap.sample())  # range >= 0
+            doc.stag(
+                "vType",
+                id=actor.id,
+                accel=actor.accel,
+                decel=actor.decel,
+                vClass=actor.vehicle_type,
+                speedFactor=actor.speed.mean,
+                speedDev=actor.speed.sigma,
+                sigma=sigma,
+                minGap=min_gap,
+                maxSpeed=actor.max_speed,
+                **actor.lane_changing_model,
+                **actor.junction_model,
+                )
```

Comment:
I think this section can be combined with the flow actor `vType` generation:

https://github.com/huawei-noah/SMARTS/blob/305b1e0c7cf829978c7efdb49cd85f3dad35b6d4/smarts/sstudio/generators.py#L215-L217

and change that to:

```
 for actor in { 
     actor for flow in traffic.flows for actor in [*flow.actors.keys(), *trip.actors.keys()]
 }: 
```